### PR TITLE
[vcpkg-baseline][zxing-cpp] Fix build failure

### PR DESCRIPTION
**Pull request by JackBoosY**
When building zxing-cpp, it links to libiconv, which depends on libcharset, but it is not added to the link list.
So add `vcpkg-cmake-wrapper.cmake` to libiconv and add the find libcharset code.
Set `opencv` and `iconv` as features.

Related: #11502.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Since FindIconv is a cmake module shipped by cmake itself the correct solution would be to install a vcpkg-cmake-wrapper.cmake for iconv.

**Comment by JackBoosY** [comment]
@Neumann-A Since libiconv now exports cmake configuration files, why not use it?

**Comment by Neumann-A** [comment]
Because it is unofficial and other cmake ports also depending on it don't benefit from it without the wrapper. In the future iconv might also be changed to use its native buildsystem instead of the custom vcpkg one

**Comment by JackBoosY** [comment]
@Neumann-A The `FindIconv.cmake` doesn't find `libcharset`, so I think we should fix it in `vcpkg-cmake-wrapper.cmake`.

File: ports/zxing-cpp/0002-improve-features.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (Neumann-A) [inline ports/zxing-cpp/0002-improve-features.patch:1]: You dont need this patch. Use `-DCMAKE_DISABLE_FIND_PACKAGE_<package>=ON` in the `vcpkg_configure_cmake` call instead (https://cmake.org/cmake/help/v3.0/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)

**Comment by JackBoosY** [comment]
Needs to move glib tombstone later,

**Comment by dan-shaw** [comment]
@JackBoosY Is this PR complete?

**Comment by JackBoosY** [comment]
@dan-shaw Yes,  merge this plz.

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**